File: platform/nexus-console/src/api/types/alerts.ts
export type AlertSeverity = 'critical' | 'high' | 'medium' | 'low' | 'informational';

/** Sensor / SIEM origin for Console filters (ADR 0011 hybrid + Wazuh full-SIEM). */
export type AlertSource =
  | 'wazuh'
  | 'suricata'
  | 'zeek'
  | 'falco'
  | 'tetragon'
  | 'ai-inference'
  | 'vector';

export interface AITriageResult {
  confidenceScore: number;
  recommendedAction: string;
  reasoningExcerpt: string;
}

export interface SOCAlert {
  id: string;
  timestamp: string;
  severity: AlertSeverity;
  source: AlertSource;
  ruleName: string;
  affectedHost: string;
  acknowledged: boolean;
  athenaScenario?: string;
  payload: Record<string, unknown>;
  /** Optional deep-link when a SIEM dashboard is deployed (full Wazuh path). */
  externalDashboardUrl?: string;
  /** @deprecated Prefer externalDashboardUrl */
  wazuhDashboardUrl?: string;
  triageResult?: AITriageResult;
}

export interface AlertFilters {
  severity?: AlertSeverity[];
  source?: AlertSource;
  timeRange?: { start: string; end: string };
}

export const ALERT_SOURCE_OPTIONS: { value: AlertSource; label: string }[] = [
  { value: 'suricata', label: 'Suricata' },
  { value: 'zeek', label: 'Zeek' },
  { value: 'falco', label: 'Falco' },
  { value: 'tetragon', label: 'Tetragon' },
  { value: 'ai-inference', label: 'AI triage' },
  { value: 'vector', label: 'Vector' },
  { value: 'wazuh', label: 'Wazuh' },
];

File: platform/api-gateway/src/models/alerts.py
"""Alert response models (camelCase for Nexus Console)."""
from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


class SOCAlert(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    timestamp: str
    severity: Literal["critical", "high", "medium", "low", "informational"]
    source: Literal["wazuh", "suricata"]
    rule_name: str = Field(alias="ruleName")
    affected_host: str = Field(alias="affectedHost")
    acknowledged: bool = False
    athena_scenario: str | None = Field(None, alias="athenaScenario")
    payload: dict[str, Any] = Field(default_factory=dict)
    wazuh_dashboard_url: str | None = Field(None, alias="wazuhDashboardUrl")


class AlertsResponse(BaseModel):
    alerts: list[SOCAlert]
    total: int


class TriageResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    confidence_score: float = Field(alias="confidenceScore")
    recommended_action: str = Field(alias="recommendedAction")
    reasoning_excerpt: str = Field(alias="reasoningExcerpt")
